### DISCUSSION TITLE
Test for Manifold Mesh

### DISCUSSION THREAD

**Discussion by adamgranthendry** [pyvista/pyvista#2626] [Q&A]
Do `pyvista` objects have any property like `is_manifold`? Per [this VTK Discourse post](https://discourse.vtk.org/t/is-there-any-way-to-check-mesh-is-watertight/3169), credit to @banesullivan, perhaps this could be achieved with:

```python
def is_manifold(mesh):
    alg = vtk.vtkFeatureEdges()

    alg.FeatureEdgesOff()
    alg.BoundaryEdgesOn()
    alg.NonManifoldEdgesOn()

    alg.SetInputDataObject(mesh)

    alg.Update()

    return alg.GetOutput().GetNumberOfCells() < 1
```

Would this be useful to add to the `pyvista` data objects?

**Comment by akaszynski** [comment]
Good idea!  Yes, we can add that.  Can you open a feature request on pyvista?
